test/e2e/dedicated_reader_test.go

```diff
@@ -0,0 +1,74 @@
+//+build e2e
+
+package e2e
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	_ "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var _ = Describe("Openshift on Azure dedicated-cluster-reader e2e tests [DedicatedAdmin]", func() {
+	defer GinkgoRecover()
+
+	BeforeEach(func() {
+		namespace := nameGen.generate("e2e-test-")
+		c.createProject(namespace)
+	})
+
+	AfterEach(func() {
+		c.cleanupProject(10 * time.Minute)
+	})
+
+	It("should read nodes", func() {
+		_, err := dreader.kc.CoreV1().Nodes().Get("master-000000", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should have read-only on all non-infrastructure namespaces", func() {
+		wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+			rb, err := dreader.kc.RbacV1().RoleBindings(c.namespace).Get("view", metav1.GetOptions{})
+			if err != nil {
+				return false, nil
+			}
+			for _, subject := range rb.Subjects {
+				if subject.Kind == "Group" && subject.Name == "dedicated-readers" {
+					return true, nil
+				}
+			}
+			return false, nil
+		})
+		// get namespace created by user
+		_, err := dreader.pc.Projects().Get(c.namespace, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		// attempt to delete namespace
+		err = dreader.pc.Projects().Delete(c.namespace, &metav1.DeleteOptions{})
+		Expect(err).To(HaveOccurred())
```

Comment:
Check that the actual error is a permission error
```go
Expect(kerrors.IsForbidden(err)).To(Equal(true))
```

Reply:
added.